src/lambda_vm/interpreter.ml

```diff
@@ -3,13 +3,16 @@ open Checks
 
 type error =
   (* interpreter bugs *)
-  | Undefined_variable
-  | Over_applied_primitives
-  (* user program bugs *)
-  | Value_is_not_pair
-  | Value_is_not_int64
-  | Value_is_not_function
-  | Value_is_not_zero
+  [ `Undefined_variable
+  | `Over_applied_primitives
```

Comment:
yeah, but that way you will always need to handle to error the error, you can always forget to handle an exception.

Reply:
The exceptions should not be handled inside of lambda_vm, they're exceptions exactly because of that.

Reply:
> yeah, but that way you will always need to handle to error the error, you can always forget to handle an exception.

The point is that `Undefined_variable` and `Over_applied_primitives` aren't supposed to be handled anyway, because they shouldn't happen in normal conditions, only if there is a bug on the interpreter itself.

Reply:
If we don't handle it inside, then there are two paths: 
1. Crash the node with unhandled error
2. Make node aware of such error and... crash the node or do nothing?

Reply:
I think that for those specific errors I've mentioned, it is worth to really just crash node, since that's not expected to happen whatsoever. They aren't even on the tests, because there's no current way of triggering them from the public API.

Reply:
If they are not expected to happen, they shouldn't even be there, it should be just ```assert false```, if we just crash the node on them, then there is a possibility that someone could find a way to abuse them to crash the nodes. So that's why for now I'm in favor of treating them as normal errors or not having them at all if we are adamant that they can't occur.

Reply:
Yeah, I guess that's ok.